eng/config/PublishData.json

```diff
@@ -1,9 +1,66 @@
 {
+  "feeds": {
+      "dotnet5": "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json",
+      "vssdk": "https://pkgs.dev.azure.com/azure-public/vside/_packaging/vssdk/nuget/v3/index.json",
+      "vs-impl": "https://pkgs.dev.azure.com/azure-public/vside/_packaging/vs-impl/nuget/v3/index.json"
+  },
+  "packages": {
```

Comment:
Listing this as a global entry means that it can never differ between our branches. Hence at any point in the future if we decide to change where these packages are published we'll be stuck because all branches will be reading this master list.

Reply:
I see - which would be better then, specifying the per branch mapping as a global entry (as we do currently for the other publishing data) or just reading this map from a place local to the currently running build?

Reply:
The mapping just needs to be flexible here. Essentially what happens in the future when we need to change `master` but not break our publishing for `release/dev16.8`. 

Think there are a couple ways to do this:

1. Just add this data into our existing branch specific entries below. That will work just fine but it does mean that our file is going to start getting big really fast. 
1. Could do some type of diffing system where this is the default and we put deviations in the branch specific entries. Thta may be overkill though.

Reply:
I ended up with something in between 1 and 2:
1.  Added keys in the packages map that can define different sets of package mappings.
2.  Updated the branch configuration to reference the correct key from the packages object.

This way it's possible to change the mappings based on branch and re-use mappings when they don't change.  Drawback is that the entire mapping has to be specified again, rather than just the diff.  Seems simple enough for now and if we end up needing to modify it a lot we can re-think this.

Reply:
I think this is a good approach.